### ISSUE TITLE
Support null bytes in message bodies

### ISSUE THREAD

**Issue by michaelklishin** [rabbitmq/rabbitmq-server#182]
See [this mailing list thread](https://groups.google.com/forum/#!topic/rabbitmq-users/OAMwU7Bfp7w). Apparently PHP serialisation function uses null bytes.

The spec says null bytes are prohibited in short strings but long strings "can contain any data."

**Comment by pinepain** [comment]
Same for `body-payload` (message body) which covered by `*OCTET` grammar.

**Comment by videlalvaro** [comment]
It seems this issue trips Postgres as well http://php.net/manual/en/function.serialize.php#96504

**Comment by videlalvaro** [comment]
We tried to reproduce the issue with **php-amqplib**, **Bunny** on ruby and **Clojure** for the JVM but we couldn't. Perhaps there's a problem on how your library encodes the data?

I even added a print statement inside rabbitmq to see what message body I was receiving. For this string in ruby: `"123\0|45"` rabbitmq prints as body: `123^@|45` which is correct. Our ruby process is able to consume the same message back.

**Comment by pinepain** [comment]
I double checked all my tests and environment and I found that I've made a stupid mistake during issue testing.

I was testing the case in which client library ([php-amqp](https://github.com/pdezwart/php-amqp)) truncating message and I used the same queue name each time I run the test case and target queue was declared as persistent. And I didn't use auto-ack and didn't acked received message. So when I made proper fix to client lib code, I still received that fist truncated message, even already sending proper message.

I'm terribly sorry that I bothered you.
